cofmupy/server/services.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+# Copyright 2025 IRT Saint Exupéry and HECATE European project - All rights reserved
+#
+# The 2-Clause BSD License
+#
+# Redistribution and use in source and binary forms, with or without modification, are
+# permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this list of
+#    conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice, this list
+#    of conditions and the following disclaimer in the documentation and/or other
+#    materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+# THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""
+Services methods for Flask server
+"""
+import os
+import json
+from cofmupy.utils import fmu_utils
+
+
+def retrieve_project_from_params(project_name: str, project_id: str, root_path: str):
+    """
+    Retrieve project from name and id. find project sub-directory of the root path,
+        and check consistency with project id
+
+    Args:
+        project_name (str): name of the project, correspond to sub-directory
+        project_id (str): project id. Used to check consistency
+        root_path (str): root path of the projects
+
+    Returns:
+        dict: config dictionary amended with additional data.
+    """
+    if not os.path.exists(os.path.join(root_path, project_name)):
+        raise Exception("Project doesn't exist")
+
+    with open(
+        os.path.join(root_path, project_name, "metadata.json"),
+        "r",
+        encoding="utf-8",
+    ) as file:
+        project = json.load(file)
+        # Check project id
+        if project["id"] != project_id:
+            raise Exception("Bad project id")
+
+    return project
+
+
+def transform_config_for_frontend(config: dict, project_path: str):
```

Comment:
I think we don't have time to apply to many changes, but I think that CoFmuPy should not have any information on how the GUI expects data format. These transformation functions could be on the GUI side: the CoFmuPy sends all necessary data in its own format, and the GUI transforms data in its own expected format. Is it not better that the GUI adapts to CoFmuPy and not the opposite?